src/components/TextEditor/components/TextEditor.js

```diff
@@ -169,28 +146,61 @@ class TextEditor extends React.Component {
           return res.json();
         })
         .then((json) => {
-          const { uid, ...programData } = json;
-          this.setState({ forking: false, forked: true });
-          this.props.addProgram(uid, programData || {});
+          const { uid2, ...programData } = json;
+          setForking(false);
+          setForked(true);
+          addProgram(uid2, programData || {});
         })
         .catch((err) => {
-          this.setState({
-            error: 'Failed to create sketch, please try again later',
-          });
-          console.log(err);
+          console.error(err);
+          throw new Error('Failed to create sketch, please try again later');
```

Comment:
@lumisphere902 is this what you expect for error?

Reply:
Hmm... we can probably just remove the throw statement.